test/cpp/naming/resolver_component_test.cc

```diff
@@ -367,5 +360,50 @@ int main(int argc, char **argv) {
   ::testing::InitGoogleTest(&argc, argv);
   grpc::testing::InitTest(&argc, &argv, true);
 
-  return RUN_ALL_TESTS();
+  grpc_init();
+
+  SubProcess *dns_server_subprocess = NULL;
+
+  if (FLAGS_start_local_dns_server == true) {
+    /* spawn a DNS server subprocess*/
+    local_dns_server_port = grpc_pick_unused_port_or_die();
+    std::string my_bin = argv[0];
+    std::string bin_dir = my_bin.substr(0, my_bin.rfind('/'));
+    std::vector<std::string> args = {
+        "tools/run_tests/python_utils/dns_server.py",
+        "--dns_port=" + std::to_string(local_dns_server_port)};
+    dns_server_subprocess = new SubProcess(args);
+
+    // Wait for the DNS server to stand up.
+    // TODO: can we get rid of the need to sleep here?
```

Comment:
This seems a bit brittle, since we don't really have any guarantee as to how long it will take the DNS server to start up.  A better approach might be to have a loop where we periodically send a query to the DNS server (maybe via "dig") and keep looping until it succeeds.  Once we've gotten a successful result from it, we know it's up.

Reply:
Good point, I will try to use dig like this to get rid of the sleep. I suppose using dig to resolve might have an added sanity checking benefit for correctness of the DNS server.